src/evolution/Evolution.cpp

```diff
@@ -366,40 +366,92 @@ DynamicMatrix assembleInverseMassMatrix(FiniteElementSpace& fes)
 	return toEigen(*bf.SpMat().ToDenseMatrix());
 }
 
-DynamicMatrix getReferenceInverseMassMatrix(const Element::Type elType, const int order, const int dimension)
+Mesh getMeshForReferenceElementBasedOnGeomType(const Element::Type elType, const int dimension)
 {
-	std::unique_ptr<Mesh> m;
 	switch (dimension) {
 	case 2:
-		m = std::make_unique<Mesh>(Mesh::MakeCartesian2D(1, 1, elType));
-		break;
+		switch (elType) {
+		case Element::Type::TRIANGLE:
+			return Mesh::MakeCartesian2D(1, 1, elType, true, 2.0, 2.0);
+		case Element::Type::QUADRILATERAL:
+			return Mesh::MakeCartesian2D(1, 1, elType);
+		default:
+			throw std::runtime_error("Incorrect Element Type for dimension 2 mesh.");
+		}
 	case 3:
-		m = std::make_unique<Mesh>(Mesh::MakeCartesian3D(1, 1, 1, elType));
-		break;
+		return Mesh::MakeCartesian3D(1, 1, 1, elType);
 	default:
 		throw std::runtime_error("Hesthaven Evolution Operator only supports dimensions 2 or 3.");
 	}
+}
+
+DynamicMatrix assembleHesthavenReferenceElementInverseMassMatrix(const Element::Type elType, const int order, const int dimension)
+{
+	auto m{ getMeshForReferenceElementBasedOnGeomType(elType, dimension) };
 	auto fec{ L2_FECollection(order, dimension, BasisType::GaussLobatto) };
-	auto fes{ FiniteElementSpace(m.get(), &fec)};
+	auto fes{ FiniteElementSpace(&m, &fec)};
 	auto mass_mat{ assembleInverseMassMatrix(fes) };
 
-	DynamicMatrix res = getElementMassMatrixFromGlobal(0, mass_mat);
-	return res;
+	return getElementMassMatrixFromGlobal(0, mass_mat);
+}
+
+DynamicMatrix assembleHesthavenReferenceElementEmat(const Element::Type elType, const int order, const int dimension)
+{
+	auto m{ getMeshForReferenceElementBasedOnGeomType(elType, dimension) };
+	m.SetAttribute(0, hesthavenMeshingTag);
+	Array<int> elementMarker;
+	elementMarker.Append(hesthavenMeshingTag);
+	auto sm{ SubMesh::CreateFromDomain(m, elementMarker) };
+	auto fec{ L2_FECollection(order, dimension, BasisType::GaussLobatto) };
+	FiniteElementSpace subFES(&sm, &fec);
+
+	auto boundary_markers = assembleBoundaryMarkers(subFES);
+
+	for (auto f{ 0 }; f < subFES.GetNF(); f++) {
+		sm.bdr_attributes[f] = f + 1;
+		sm.SetBdrAttribute(f, sm.bdr_attributes[f]);
+	}
+
+	return assembleEmat(subFES, boundary_markers);
 }
 
-GlobalBoundaryMap assembleGlobalBoundaryMap(BoundaryToMarker& markers, FiniteElementSpace& fes)
+GlobalBoundary assembleGlobalBoundaryMap(const GlobalConnectivity& vmaps, BoundaryToMarker& markers, FiniteElementSpace& fes)
 {
-	GlobalBoundaryMap res;
+	GlobalBoundary res;
```

Comment:
Would ExternalBoundary be more descriptive and show a dualism with InteriorBoundary?

Reply:
The method itself is deprecated and I hadn't removed it, boundary connectivity is done through other methods now and the definition of GlobalBoundary is not used anymore (Neither is InteriorBoundary at this point).